### PR TITLE
Tooltip Update

### PR DESCRIPTION
**Pull request by xizi-xu**
1. Fix the issue of mouseover circles placed outside of the container.
2. Move x-axis tag (time) to the top of the graph.
3. Sort sources in legend alphabetically.
screen shot 2016-12-06 at 4 11 55 pm

4. Remove time value in the tooltip.
5. Add sorting by value feature for tooltip items.
screen shot 2016-12-06 at 4 12 09 pm

6. Add detail tooltip option, which displays source names and more precision for data values.
screen shot 2016-12-06 at 4 11 23 pm

### CODE REVIEW

File: ArgusWeb/app/js/directives/charts/chart.js [hunk 1]
```diff
@@ -25,6 +25,12 @@ function(Metrics, Annotations, ChartRenderingService, ChartDataProcessingService
             // use graphClassName to bind all the graph element of a metric together
             lineChartScope.series[i].graphClassName = newChartId + "_graph" + (i + 1);
         }
+        // sort series alphabetically
+        lineChartScope.series = lineChartScope.series.sort(function(a, b) {
+            var textA = a.name.toUpperCase();
+            var textB = b.name.toUpperCase();
+            return (textA < textB) ? -1 : (textA > textB) ? 1 : 0;
```
Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/chart.js]: Nice!

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 2]
```diff
@@ -1087,7 +1088,7 @@ angular.module('argus.directives.charts.lineChart', [])
                 graphClassNames = series.map(function(metric) { return metric.graphClassName; });
                 legendCreator(names, colors, graphClassNames);
                 // check if there is anything to graph
-                var hasNoData, emptyReturn, invalidExpression, haveThingsToGraph;
```
Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: Aww, I was getting used to that.

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 3]
```diff
@@ -405,61 +408,93 @@ angular.module('argus.directives.charts.lineChart', [])
...
                 var XOffset = 0;
                 var YOffset = 0;
+                var newXOffset = 0;
                 var OffsetMultiplier = -1;
                 var itemsPerCol = 8;
                 var circleLen = circleRadius * 2;
+                if (scope.menuOption.isTooltipDetailOn) {
+                    itemsPerCol = 14;
+                } else if (datapoints.length < 2*itemsPerCol) {
+                    itemsPerCol = Math.ceil(datapoints.length / 2);
```
Comment (prestonfff) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: Nice handling of the detail.

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 4]
```diff
@@ -405,61 +408,93 @@ angular.module('argus.directives.charts.lineChart', [])
...
+                                        .attr('cy', 20 * (0.75 + i - YOffset) + Y)
+                                        .attr('cx', X + tipOffset + tipPadding + circleRadius + XOffset);
+                    var textLine = group.select("text." + datapoints[i].graphClassName)
+                                        .attr('dy', 20 * (1 + i - YOffset) + Y)
+                                        .attr('dx', X + tipOffset + tipPadding + circleLen + 2 + XOffset);
+
+                    if (scope.menuOption.isTooltipDetailOn) {
+                        textLine.text(datapoints[i].name + "   " + d3.format('0,.7')(tempData));
+                    } else {
+                        textLine.text(d3.format('.2s')(tempData));
```
Comment (prestonfff) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: Maybe add some comma for the number? like 999,999,999

Comment (xizi-xu) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: ',' in d3.format is adding comma.

**Pull request merged**